doc/how_to_introduce_breaking_changes.md

```diff
@@ -0,0 +1,237 @@
+
+```python
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from azure.cli.core.breaking_change import upcoming_breaking_changes
+```
+
+Then you could pre-announce breaking changes for different command groups or command, both list and `BreakingChange` object are accepted.
+
+```python
+from azure.cli.core.breaking_change import upcoming_breaking_changes, AzCLIBeRequired, AzCLIDefaultChange,
+  AzCLIOtherChange
+
+upcoming_breaking_changes['bar foo'] = AzCLIBeRequired('--name')
+upcoming_breaking_changes['bar foo baz'] = [AzCLIDefaultChange('--foobar', 'A', 'B'), AzCLIOtherChange(
+  'During May 2024, another Breaking Change would happen in Build Event.')]
+```
+
+All related breaking changes would be displayed while executing the command. For example, in the above 
+
+```shell
+# The azure command
+az bar foo baz
+
+# =====Warning output=====
+# The argument `--name` will become required in next breaking change release(2.61.0).
+# The default value of `--foobar` will be changed to `B` from `A` in next breaking change release(2.61.0).
+# During May 2024, another Breaking Change would happen in Build Event.
+```
+
+There are several types of breaking changes defined in `breaking_change`. You should use any of them to declare breaking changes in `_breaking_change.py`.
+
+**Remove**
+
+```python
+from azure.cli.core.breaking_change import upcoming_breaking_changes, AzCLIRemoveChange, NextBreakingChangeWindow
+
+# Remove the command groups, commands or arguments in a future release.
+# **It is recommended to utilize `deprecate_info` instead of this class to pre-announce Breaking Change of Removal.**
+upcoming_breaking_changes['bar foo'] = AzCLIRemoveChange('az bar foo', target_version=NextBreakingChangeWindow(),
+                                                         redirect='`az barfoo`')
+# `az bar foo` will be removed in next breaking change release(2.61.0). Please use `az barfoo` instead.
+```
+
+**Rename**
+
+```python
+from azure.cli.core.breaking_change import upcoming_breaking_changes, AzCLIRenameChange, NextBreakingChangeWindow
+
+# Rename the command groups, commands or arguments to a new name in a future release.
+# **It is recommended to utilize `deprecate_info` instead of this class to pre-announce Breaking Change of Renaming.**
+# It is recommended that the old name and the new name should be reserved in few releases.
+upcoming_breaking_changes['bar foo'] = AzCLIRenameChange('az bar foo', 'az bar baz',
+                                                         target_version=NextBreakingChangeWindow())
+# `az bar foo` will be renamed to `az bar baz` in next breaking change release(2.61.0).
+```
+
+**OutputChange**
```

Comment:
```suggestion
**Output Change**
```

Reply:
My personal suggestion is to have spaces between words, and this is also recommended for similar situations below